### ISSUE TITLE
Tweet with media are failing in self hosted version

### ISSUE THREAD

**Issue by j0k3r** [itteco/iframely#498]
I can't find when it started to fail (maybe a change at Twitter side) but when a tweet contains a media, the fetch is failing.

I've tested locally with `http://0.0.0.0:8061/iframely?uri=https://twitter.com/RockoPeppe/status/582323285825736704` (using the tweet from inside `twitter.status.js`).

Here are log from the server:

```
$ node server
Using cache engine: no-cache
No local domains file detected...
-- [23-05-09 11:26:19]:16574 Loading domains list from https://iframely.com/qa/domains.json
-- [23-05-09 11:26:20]:16574 Domains list activated. 1950 domains, including disabled ones
Invalid mixin "ld-newsarticle-logo" in plugin "sverigesradio.se"
Invalid mixin "ld-author" in plugin "sverigesradio.se"
Invalid mixin "ld-date" in plugin "sverigesradio.se"
Iframely plugins loaded:
   - custom domains: 146
   - generic & meta: 95

Starting Iframely...
Base URL for embeds that require hosted renders: http://localhost:8061

 - support@iframely.com - if you need help
 - twitter.com/iframely - news & updates
 - github.com/itteco/iframely - star & contribute

iframely is running on 0.0.0.0:8061
API endpoints: /oembed and /iframely; Debugger UI: /debug

-- [23-05-09 11:26:22]:16574 127.0.0.1 - Loading /iframely for https://twitter.com/RockoPeppe/status/582323285825736704
-- [23-05-09 11:26:22]:16574    -- plugin redirect (by "htmlparser") /
-- [23-05-09 11:26:23]:16574    -- plugin response: {"plugin":"htmlparser","response":"maximum redirect reached at: https://twitter.com/RockoPeppe/status/582323285825736704","uri":"https://twitter.com/RockoPeppe/status/582323285825736704"}
Requested page error: FetchError: maximum redirect reached at: https://twitter.com/RockoPeppe/status/58232328582573670
```

And the response is:

```json
{
  "error": {
    "source": "iframely",
    "code": 417,
    "message": "Requested page error: FetchError: maximum redirect reached at: https://twitter.com/RockoPeppe/status/582323285825736704"
  }
}
```

I tried to find the _root_ issue but I failed.
The only thing that fixed the problem, is commenting that line https://github.com/itteco/iframely/blob/c38710d632f9d61136d675d3a5cf35026b1c590f/plugins/domains/twitter.com/twitter.status.js#L72

**Comment by iparamonau** [comment]
The plugin is trying to get a thumbnail image if it sees that there is one in the tweet:

https://github.com/itteco/iframely/blob/c38710d632f9d61136d675d3a5cf35026b1c590f/plugins/domains/twitter.com/twitter.status.js#L70-L74

Around when Twitter announced their monetization attempt for their developer APIs couple months ago, they also added stricter bot restrictions, hence you probably get blocked. The options we set when we fetch the URL are actually there to prevent the plugin from failing on bot rejections. 

When you remove `options.followHTTPRedirect`, the second option kicks in (`exposeStatusCode`), and basically you get a `3xx` redirect code instead. You probably don't get a picture either.

The proper fix would be to disable that fetch altogether via a feature flag. But before we do, it would be interesting to understand what Twitter gives you. Could you perhaps try to cURL that URL and follow the redirection chain manually and see what happens?

**Comment by iparamonau** [comment]
Thank you for the details. We actually could repeat similar issue with Twitter users and timelines. It is likely because of Fetch module that we currently use does not re-use cookies between redirects by the default. We have used our jars elsewhere in the code, but not for Twitter (due to `followHTTPredirect` bypass used). 

The issue is resolved in #502. Also, we now have [disabled Twitter thumbnail](https://github.com/itteco/iframely/commit/5a939faa513d982f668d534e264e4ac78e4586f9) by default to avoid issues in the future. To enable, please configure `twitter: {thumbnail: true, ...}` in your `providerOptions` config.
